components/public-api/typescript-common/src/prebuild-utils.ts

```diff
@@ -17,6 +17,30 @@ export function getPrebuildLogPath(prebuildId: string): string {
     return PREBUILD_LOGS_PATH_PREFIX + "/" + prebuildId;
 }
 
+/** cmp. @const HEADLESS_LOG_STREAM_ERROR_REGEX */
+const PREBUILD_LOG_STREAM_ERROR = "X-Prebuild-Error";
+const PREBUILD_LOG_STREAM_ERROR_REGEX = /X-Prebuild-Error#(?<code>[0-9]+)#(?<message>.*?)#X-Prebuild-Error/;
+
+export function matchPrebuildError(msg: string): undefined | ApplicationError {
+    const result = PREBUILD_LOG_STREAM_ERROR_REGEX.exec(msg);
+    if (!result || !result.groups) {
+        return;
+    }
+    return new ApplicationError(Number(result.groups.code) as ErrorCode, result.groups.message);
+}
+
+export function getPrebuildErrorMessage(err: any) {
+    let code: ErrorCode = ErrorCodes.INTERNAL_SERVER_ERROR;
+    let message = "unknown error";
+    if (err instanceof ApplicationError) {
+        code = err.code;
+        message = err.message;
+    } else if (err instanceof Error) {
+        message = err.message;
```

Comment:
Should we expose real error message?

Reply:
What does a "real error message" mean? Is it not `.message`?

Reply:
@filiptronicek  error message may leak sensitive data or info's that user don't need to know. But sometimes we also need to know if it's failed because of fetch failed

Reply:
Maybe we just need to omit them all, and wrap into ApplicationError once that place necessary

Reply:
I think if we log server-side, it's most important. If there is some complex issue, it is indeed probably not helpful to show it.